File: investing-backend/src/main/java/com/davidmuns/investing/dto/PositionCloseResponse.java
package com.davidmuns.investing.dto;

import java.time.LocalDate;

public record PositionCloseResponse(
        Long id,
        String name,
        String symbol,
        LocalDate openedAt,
        String type,
        Double quantity,
        Double entryPrice,
        LocalDate closedAt,
        Double closedPrice,
        Double profitLossPercentage,
        Double profitLoss,
        Long portfolioId
) {
}

File: investing-frontend/src/app/shared/models/position-close-response.ts
export interface PositionCloseResponse {
  id: number;
  name: string;
  symbol: string;
  openeddAt: string;
  type: string;
  quantity: number;
  entryPrice: number;
  closeddAt: string;
  closedPrice: number;
  profitLossPercentage: number;
  profitLoss: number;
  portfolioId: number;
}
